lifetimes/estimation.py

```diff
@@ -37,6 +37,109 @@ def _print_params(self):
         return s.strip(', ')
 
 
+class GammaGammaFitter(BaseFitter):
+    def __init__(self, penalizer_coef=0.):
+        self.penalizer_coef = penalizer_coef
+
+    @staticmethod
+    def _negative_log_likelihood(params, frequency, avg_monetary_value, penalizer_coef=0):
+        if any(i < 0 for i in params):
+            return np.inf
+
+        p, q, v = params
+
+        x = frequency
+        m = avg_monetary_value
+
+        negative_log_likelihood_values = special.gammaln(p*x+q)-special.gammaln(p*x)-special.gammaln(q)\
+                                         +q*np.log(v)+(p*x-1)*np.log(m)+(p*x)*np.log(x)-(p*x+q)*np.log(x*m+v)
+
+        penalizer_term = penalizer_coef * log(params).sum()
+        negative_log_likelihood = -np.sum(negative_log_likelihood_values) + penalizer_term
+
+        return negative_log_likelihood
+
+    def conditional_expected_average_profit(self, x=None, m=None):
+        """
+        This method computes the conditional expectation of the average profit per transaction
+        for a group of one or more customers.
+            x: a vector containing the customers' frequencies. Defaults to the whole set of
+                frequencies used for fitting the model.
+            m: a vector containing the customers' monetary values. Defaults to the whole set of
+                monetary values used for fitting the model.
+
+        Returns:
+            the conditional expectation of the average profit per transaction
+        """
+        m = self.data['monetary_value'] if m is None else m
+        x = self.data['frequency'] if x is None else x
+        p, q, v = self._unload_params('p', 'q', 'v')
+        return np.mean((((q - 1) / (p * x + q - 1)) * (v * p / (q - 1))) + (p * x / (p * x + q - 1)) * m)
```

Comment:
Also, does it make more sense for this to be applied row-wise. ATM, if I enter: 
```
ggf.conditional_expected_average_profit(returning_customers_summary['frequency'].ix[:10], returning_customers_summary['monetary_value'].ix[:10])
```

I get a single value back. I feel like I should instead get the `conditional_expected_average_profit` per customer, essentially returning an array.  What do you think?